activerecord/test/cases/migration/exclusion_constraint_test.rb

```diff
@@ -172,6 +172,14 @@ def test_added_deferrable_initially_immediate_exclusion_constraint
           end
         end
 
+        def test_exclusion_constraint_exists
```

Comment:
What do you think about adding some tests using `:expression`?

Reply:
I deliberately excluded checking expressions, because it is too brittle. That is currently skipped for check constraints, for example https://github.com/rails/rails/blob/fc75fed35a90ee7edc7a9aed0cec663928467e4d/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb#L188-L192